### PR TITLE
Update to build with CUDA 11.7.

### PR DESCRIPTION
**Pull request by bdice**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Rebuilding to allow `__cuda` pinning to match the recently released CUDA 11.7. I also added myself to the recipe maintainers, if that's alright with you @prashantkhoje!

cc: @quasiben @gmarkall

### CODE REVIEW

**Comment by bdice** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -49,3 +49,4 @@ about:
 extra:
   recipe-maintainers:
   - prashantkhoje
+  - bdice
```
Comment (bdice) [inline recipe/meta.yaml:52]: @jakirkham requested to be added to the maintainer list as well. @jakirkham please confirm in a comment.
```suggestion
  - bdice
  - jakirkham
```

**Comment by bdice** [comment]
@conda-forge-admin, please rerender

**Comment by gmarkall** [comment]
We'd talked before about removing the upper bound on the `__cuda` runtime constraint here:

https://github.com/conda-forge/ptxcompiler-feedstock/blob/a2ff3370ed836c30f4e65e6a7677365262b024ca/recipe/meta.yaml#L36

Does that need doing for this PR too?

**Comment by prashantkhoje** [comment]
If builds with 11.x can run in any 11.x environment, i agree we can remove upper bound.
 
I wonder why the build failed to run for ppc64le python 3.9.

**Comment by bdice** [comment]
> We'd talked before about removing the upper bound on the `__cuda` runtime constraint here:
> 
> https://github.com/conda-forge/ptxcompiler-feedstock/blob/a2ff3370ed836c30f4e65e6a7677365262b024ca/recipe/meta.yaml#L36
> 
> Does that need doing for this PR too?

I wasn't 100% sure if you wanted to do that or not. I will remove it.

**Comment by prashantkhoje** [comment]
Thank you for the contribution.

**Comment by jakirkham** [comment]
Thanks Prashant! 🙏

**Comment by bdice** [comment]
Yes, thank you!

**Pull request merged**